ui/src/app/features/face-recognition/face-recognition-container.component.ts

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright (c) 2020 the original author or authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+import { Component, Input, OnDestroy, OnInit } from '@angular/core';
+import { combineLatest, Observable } from 'rxjs';
+import { map } from 'rxjs/operators';
+import { Store } from '@ngrx/store';
+import { AppState } from '../../store';
+import { recognizeFace, recognizeFaceReset } from '../../store/face-recognition/actions';
+import { selectFaceData, selectFile, selectRequest,
+  selectTestIsPending } from '../../store/face-recognition/selectors';
+
+@Component({
+  selector: 'app-face-recognition-container',
+  templateUrl: './face-recognition-container.component.html',
+  styleUrls: ['./face-recognition-container.component.scss']
+})
+export class FaceRecognitionContainerComponent implements OnInit, OnDestroy {
+  data$: Observable<any>;
+  file$: Observable<any>;
+  requestInfo$: Observable<any>;
+  pending$: Observable<boolean>;
+  isDisplayResult$: Observable<boolean>;
+
+  @Input()
+  apiKey: string;
```

Comment:
please refactor and remove apiKey as an input. Use `withLatestFrom` RxJs operator in effects for `recognizeFace` action, to select correct apiKey